File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque
n, m, v = map(int, input().split())
graph = [[False]*(n+1) for _ in range(n+1)]

for i in range(m):
    a, b = map(int, input().split())
    graph[a][b] = True
    graph[b][a] = True

visited_dfs = [False]*(n+1)
visited_bfs = [False]*(n+1)

def dfs(v):
    visited_dfs[v] = True
    print(v, end = " ")
    for i in range(1, n+1):
        if not visited_dfs[i] and graph[v][i]:
            dfs(i)
    
def bfs(v):
    q = deque([v])
    visited_bfs[v] = True
    while q:
        v = q.popleft()
        print(v, end = " ")
        for i in range(1, n+1):
            if not visited_bfs[i] and graph[v][i]:
                q.append(i)
                visited_bfs[i] = True
    

dfs(v)
print()
bfs(v)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main {
    static int n, m, v;
    static boolean[][] graph;
    static boolean[] visited_bfs;
    static boolean[] visited_dfs;

    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(bf.readLine());

        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        v = Integer.parseInt(st.nextToken());

        graph = new boolean[n + 1][n + 1];
        visited_bfs = new boolean[n + 1];
        visited_dfs = new boolean[n + 1];

        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(bf.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());

            graph[a][b] = true;
            graph[b][a] = true;
        }

        dfs(v);
        System.out.println();
        bfs(v);
    }

    static void dfs(int v) {
        visited_dfs[v] = true;
        System.out.print(v + " ");

        for (int i = 1; i <= n; i++) {
            if (!visited_dfs[i] && graph[v][i]) {
                dfs(i);
            }
        }
    }

    static void bfs(int v) {
        Queue<Integer> q = new ArrayDeque<>();
        q.add(v);
        visited_bfs[v] = true;

        while (!q.isEmpty()) {
            int x = q.poll();
            System.out.print(x + " ");

            for (int i = 1; i <= n; i++) {
                if (!visited_bfs[i] && graph[x][i]) {
                    q.add(i);
                    visited_bfs[i] = true;
                }
            }
        }
    }
}
